### PR TITLE
Handle structured core errors better

### PR DESCRIPTION
**Pull request by dividedmind**
Some code is taken from v5 branch of conjur-api.
[#83604556]

### CODE REVIEW

**Comment by kgilpin** [comment]
Thanks for this. I have a question about cli.rb. I notice that we are printing the exception to Conjur.log. This is also done by GLI_DEBUG=true. I'm not sure how the switching of the standard error handling on and off will affect this.

File: lib/conjur/cli.rb [hunk 1]
```diff
@@ -121,19 +121,26 @@ def init!
...
+          exception = err
+          $stderr.puts "error: " + err.message
+          use_default_handler = false
+        else
+          $stderr.puts exception.response.body
         end
       end
       require 'conjur/log'
       if Conjur.log
         Conjur.log << "error: #{exception}\n#{exception.backtrace.join("\n") rescue 'NO BACKTRACE?'}"
```
Comment (kgilpin) [inline lib/conjur/cli.rb]: Backtrace is also printed when `GLI_DEBUG=true`. However, I am not sure if turning `use_default_handler` off will suppress that. What does the default handler do?

Comment (kgilpin) [inline lib/conjur/cli.rb]: In other words, I think this flow could use some comments about what's happening and why :-)

**Comment by kgilpin** [comment]
Did you get a chance to run this against a live appliance?

**Comment by dividedmind** [comment]
I did, I tested against my sandbox.

Thanks for the comments. I think printing of backtrace is part of the
standard error handler; I was trying to keep the current behaviour without
giving much thought as to whether it makes sense. Perhaps you're right that
it'd be better be simplified. I'll look into it.

15:28 czw., 12.02.2015 Kevin Gilpin użytkownik <notifications@github.com>
napisał:

**Comment by dividedmind** [comment]
So, the default handler prints `"error: #{exception.message}"` to stdout, so basically what I do here (except for the original exception). With `GLI_DEBUG=true`, backtrace is additionally printed, regardless of whether the default handler is used.

I think given the above using `Conjur.log` to log the error is redundant. Ok to remove?

**Comment by kgilpin** [comment]
Yes. We probably didn't know about GLI_DEBUG

**Pull request merged**